cpp/src/parquet/encoding_test.cc

```diff
@@ -867,158 +902,71 @@ TEST_F(DictEncoding, CheckDecodeIndicesNoNulls) {
-      ASSERT_EQ(assembled_value_as_float, decoded_data[j]);
+      decoder->SetData(num_values_, encode_buffer_->data(),
+                       static_cast<int>(encode_buffer_->size()));
+      int values_decoded = decoder->Decode(decode_buf_, num_values_);
+      ASSERT_EQ(num_values_, values_decoded);
+      ASSERT_NO_FATAL_FAILURE(VerifyResults<T>(decode_buf_, draws_, num_values_));
     }
   }
-}
 
-// Check that an encode-decode pipeline produces the original small input.
-// This small-input test is added to ease debugging in case of changes to
-// the encoder/decoder implementation.
-TEST(ByteStreamSplitEncodeDecode, SmallInput) {
-  const float data[] = {-166.166f, -0.2566f, .0f, 322.0f, 178888.189f};
-  const int num_values = sizeof(data) / sizeof(data[0U]);
-  TestByteStreamSplitRoundTrip<FloatType>(data, num_values);
-}
-
-TEST(ByteStreamSplitEncodeDecode, PutSpaced) {
-  const float data[] = {-1.0f, .0f,       .0f, 3.0f,          .0f,       22.1234f,
-                        .0f,   198891.0f, .0f, -223345.4455f, 24443.124f};
-  const float valid_data[] = {-1.0f,     3.0f,          22.1234f,
-                              198891.0f, -223345.4455f, 24443.124f};
-  // The valid ones are the ones which are non-zero.
-  // The enable bits are: 10010101 011.
-  const uint8_t valid_bits[2] = {0xA9U, 0x6U};
-  const int num_values = sizeof(data) / sizeof(data[0U]);
-  const int num_valid_values = sizeof(valid_data) / sizeof(valid_data[0U]);
-  std::unique_ptr<TypedEncoder<FloatType>> encoder =
-      MakeTypedEncoder<FloatType>(Encoding::BYTE_STREAM_SPLIT);
-  encoder->PutSpaced(data, num_values, valid_bits, 0);
-  std::shared_ptr<Buffer> buffer = encoder->FlushValues();
-
-  TestByteStreamSplitDecodePath<FloatType>(buffer->data(), buffer->size(), valid_data,
-                                           num_valid_values, false);
-}
+ protected:
+  USING_BASE_MEMBERS();
+};
 
-TEST(ByteStreamSplitEncodeDecode, PutArrow) {
-  arrow::random::RandomArrayGenerator rag{1337};
-  const int num_values = 123;
-  auto arr = rag.Float32(num_values, -2048.0f, 2048.0f, 0);
-  std::unique_ptr<TypedEncoder<FloatType>> encoder =
-      MakeTypedEncoder<FloatType>(Encoding::BYTE_STREAM_SPLIT);
-  encoder->Put(*arr);
-  std::shared_ptr<Buffer> buffer = encoder->FlushValues();
-
-  auto raw_values = checked_cast<const arrow::FloatArray&>(*arr).raw_values();
-  TestByteStreamSplitDecodePath<FloatType>(buffer->data(), buffer->size(), raw_values,
-                                           num_values, false);
-}
+typedef ::testing::Types<FloatType, DoubleType> ByteStreamSplitTypes;
+TYPED_TEST_CASE(TestByteStreamSplitEncoding, ByteStreamSplitTypes);
 
-// Test that the encode-decode pipeline can handle big 32-bit FP input.
-TEST(ByteStreamSplitEncodeDecode, BigInputFloat) {
-  TestEncodeDecodeWithBigInput<FloatType>();
+TYPED_TEST(TestByteStreamSplitEncoding, BasicRoundTrip) {
+  ASSERT_NO_FATAL_FAILURE(this->Execute(10000, 1));
```

Comment:
10000 is perhaps a bit large. Some debug builds are slow (especially under Valgrind).

Reply:
Changed to 1000.